### PR TITLE
aactivator: init at 2.0.0

### PR DESCRIPTION
**Pull request by keller00**
[aactivator](https://github.com/Yelp/aactivator) automatically activate Python virtualenvs (and other environments).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
Welcome to nixpkgs! Please split adding maintainers and adding packages into two commits

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -215,6 +215,8 @@ with pkgs;
 
   # addDriverRunpath is the preferred package name, as this enables
   # many more scenarios than just opengl now.
+  aactivator = python3Packages.callPackage ../development/python-modules/aactivator { };
+
```
[the remark this answers is not in the document]
Comment (keller00) [reply pkgs/top-level/all-packages.nix]: I assume that you didn't want me to remove this, but update it once I moved the definition into `pkgs/by-name`

[the remark this answers is not in the document]
Comment (keller00) [reply pkgs/top-level/all-packages.nix]: Nvm, I see how `by-name` works!

**Comment by keller00** [comment]
@aucub  Is the normal workflow usually that people apply the suggestions manually and then force-push over their commits every time? Or is it possible to accept suggestions and not force-push over my old commits.
The commit guidelines do not make it sound like adding suggestion commits is encouraged.

Hopefully I got everything now!

File: pkgs/by-name/aa/aactivator/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "aactivator";
+  version = "2.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Yelp";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/aa/aactivator/package.nix:14]:
```suggestion
    repo = "aactivator";
```

File: pkgs/by-name/aa/aactivator/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    python3Packages.pexpect
+  ];
+
+  disabledTestPaths = [
+    # presumably because of shell manipulations
+    "tests/integration_test.py"
+  ];
+
+  meta = {
+    description = "Automatically activate Python virtualenvs (and other environments).";
```
Comment (Aleksanaa) [inline pkgs/by-name/aa/aactivator/package.nix:35]:
```suggestion
    description = "Automatically activate Python virtualenvs (and other environments)";
```

File: pkgs/by-name/aa/aactivator/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    homepage = "https://github.com/Yelp/aactivator";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ keller00 ];
```
Comment (Aleksanaa) [inline pkgs/by-name/aa/aactivator/package.nix:38]: Missing `mainProgram`

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**